src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5653,6 +5653,33 @@ static QVariant fcnLineInterpolatePoint( const QVariantList &values, const QgsEx
   return result;
 }
 
+static QVariant fcnLineInterpolatePointByM( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry lineGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  const double m = QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent );
+  const bool use3DDistance = values.at( 2 ).toBool();
+
+  double x, y, z, distance;
+
+  const QgsLineString *line = qgsgeometry_cast<const QgsLineString *>( lineGeom.constGet() );
+  if ( !line )
+  {
+    return QVariant();
```

Comment:
Shakll we not set an evaluation error string?

Reply:
I think we should match the behavior to the pre-existing line_interpolate_point(), which does not return an evaluation error string.